devicetypes/smartthings/neo-coolcam-light-switch.src/neo-coolcam-light-switch.groovy

```diff
@@ -0,0 +1,164 @@
+			 isComponent   : false, componentName: "ch2", componentLabel: "${componentLabel}"])
+		log.debug "Endpoint 2 (Binary Switch Endpoint) added as $componentLabel"
+	} catch (e) {
+		log.warn "Failed to add endpoint 2 ($desc) as Binary Switch Endpoint - $e"
+	}
+	configure()
+}
+
+def updated() {
+	configure()
+}
+
+def configure() {
+	// Device-Watch simply pings if no device events received for checkInterval duration of 32min = 2 * 15min + 2min lag time
+	sendEvent(name: "checkInterval", value: 30 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	response([zwave.configurationV1.configurationSet(parameterNumber: 4, size: 1, configurationValue: [1]).format(),
+			  "delay 100",
+			  zwave.basicV1.basicGet().format()])
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description, [0x20: 1, 0x84: 1, 0x98: 1, 0x56: 1, 0x60: 3])
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug("'$description' parsed to $result")
+	return createEvent(result)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	[name: "switch", value: cmd.value ? "on" : "off"]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x32: 3, 0x25: 1, 0x20: 1])
+	if (cmd.sourceEndPoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	} else { // sourceEndPoint == 2
+		childDevices[0]?.handleZWave(encapsulatedCommand)
+		[:]
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	def versions = [0x31: 2, 0x30: 1, 0x84: 1, 0x9C: 1, 0x70: 2]
+	def version = versions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	[descriptionText: "$device.displayName: $cmd", isStateChange: true]
+}
+
+def on() {
+	delayBetween([
+		zwave.switchBinaryV1.switchBinarySet(switchValue: 0xFF).format(),
+		zwave.switchBinaryV1.switchBinaryGet().format()
+	])
+}
+
+def off() {
+	delayBetween([
+		zwave.switchBinaryV1.switchBinarySet(switchValue: 0x00).format(),
+		zwave.switchBinaryV1.switchBinaryGet().format()
+	])
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	refresh()
+}
+
+def refresh() {
+	zwave.switchBinaryV1.switchBinaryGet().format()
+}
+
+// sendCommand is called by endpoint 2 child device handler
+def sendCommand(endpointDevice, commands) {
```

Comment:
Yes. I wanted binary switch endpoint to be reusable. This argument may be required for any multichannel device with more than one child device to identify calling endpoint.

I can try to add some code to make endpoint aware of its channel number, so It could pass number as argument. 

This will add some copy-pasting to future parent devices or a verbose comment explaining what needs to be done in parent dth for this to work.

Reply:
Last option will require type checking in any future parent device that uses more endpoint types (which is unlikely but possible).

Reply:
Last options would be to leave it be and add comments, add some redundant endpoint identification logic, or remove this argument, making endpoint dth dedicated to this device in the process.

Reply:
The dth is called `neo-coolcam-light-switch.groovy`. If you want it to be reused, you would want to change the name.

Reply:
Never mind. I get it. Looks good.